wp-content/civi-extensions/goonjcustom/Civi/CollectionCampOutcomeService.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Civi;
+
+use Civi\Api4\Contact;
+use Civi\Api4\EckEntity;
+
+/**
+ *
+ */
+class CollectionCampOutcomeService {
+
+  /**
+   * Process camp reminder logic.
+   *
+   * @param array $camp
+   * @param \DateTimeImmutable $now
+   *
+   * @throws \Exception
+   */
+  public static function processCampReminder($camp, $now, $from) {
+    $campAttendedById = $camp['Logistics_Coordination.Camp_to_be_attended_by'];
+    $endDateString = $camp['Collection_Camp_Intent_Details.End_Date'];
+    $endDate = new \DateTime($camp['Collection_Camp_Intent_Details.End_Date']);
+    $collectionCampId = $camp['id'];
+    $campCode = $camp['title'];
+    $campAddress = $camp['Collection_Camp_Intent_Details.Location_Area_of_camp'];
+
+    $lastReminderSent = $camp['Camp_Outcome.Last_Reminder_Sent'] ? new \DateTime($camp['Camp_Outcome.Last_Reminder_Sent']) : NULL;
+
+    // Calculate hours since camp ended.
+    $hoursSinceCampEnd = abs($now->getTimestamp() - $endDate->getTimestamp()) / 3600;
+
+    // Calculate hours since last reminder was sent (if any)
+    $hoursSinceLastReminder = $lastReminderSent ? abs($now->getTimestamp() - $lastReminderSent->getTimestamp()) / 3600 : NULL;
+
+    // Check if the outcome form is not filled and send the first reminder after 48 hours of camp end.
+    if ($hoursSinceCampEnd < 48) {
+      return FALSE;
+    }
+
+    // Send the reminder email if the form is still not filled.
```

Comment:
This comment does not look right.
```suggestion
    // Return if 24 hours have not passed since the last reminder.
```